src/seeds.token.cpp

```diff
@@ -193,33 +222,51 @@ void token::check_limit(const name& from) {
   check(current < limit, "too many outgoing transactions");
 }
 
-void token::resetweekly() {
-  require_auth(get_self());
-  
+void token::reset_weekly_aux(uint64_t begin) {
+
+  config_tables config(contracts::settings, contracts::settings.value);
+
+  auto batch_size = config.get(name("batchsize").value, "The batchsize parameter has not been initialized yet.");
   auto sym_code_raw = seeds_symbol.code().raw();
+  uint64_t counter = 0;
 
   transaction_tables transactions(get_self(), sym_code_raw);
 
   auto titr = transactions.begin();
-  while (titr != transactions.end()) {
+  std::advance(titr, begin);
```

Comment:
This will not work since "advance" has linear time, it's the same as iterating through all accounts

You need to use the same methods we are using in scores refactoring where the begin value is the actual account name value (not the counter)

This is in the branch https://github.com/JoinSEEDS/seeds-smart-contracts/pull/84

Reply:
Sample implementation here in rankrep - it's always the same pattern, we avoid all counting - advance, next, all these are linear time and will fail with > 1000 table entries.

We only use find and get on the table, which are logN time.

```
void accounts::rankreps() {
  rankrep(0, 0, 200);
}

void accounts::rankrep(uint64_t start_val, uint64_t chunk, uint64_t chunksize) {
  require_auth(_self);

  // DEBUG REMOVE THIS - THIS IS SO THE FUNCTION DOESN"T RUN AMOK
  //if (chunk > 10) { 
  //  return;
  //}

  uint64_t total = get_size("rep.sz"_n);
  uint64_t current = chunk * chunksize;
  auto rep_by_rep = rep.get_index<"byrep"_n>();
  auto ritr = start_val == 0 ? rep_by_rep.begin() : rep_by_rep.lower_bound(start_val);
  uint64_t count = 0;

  while (ritr != rep_by_rep.end() && count < chunksize) {

    uint64_t rank = (current * 100) / total;

    rep_by_rep.modify(ritr, _self, [&](auto& item) {
      item.rank = rank;
    });

    current++;
    count++;
    ritr++;
  }

  if (ritr == rep_by_rep.end()) {
    // Done.
  } else {
    // recursive call
    uint64_t next_value = ritr->by_rep();
    action next_execution(
        permission_level{get_self(), "active"_n},
        get_self(),
        "rankrep"_n,
        std::make_tuple(next_value, chunk + 1, chunksize)
    );

    transaction tx;
    tx.actions.emplace_back(next_execution);
    tx.delay_sec = 1;
    tx.send(next_value + 1, _self);
    
  }

}
```

Reply:
Hello, I committed the change. I removed the `advance` and I used `lower_bound` instead as you suggested.

Reply:
Perfect